### ISSUE TITLE
Status text returned by a ZAP handler is masked

### ISSUE THREAD

**Issue by sigiesec** [zeromq/libzmq#2653]
The status text returned by a ZAP handler, particularly for status codes other than 200, is masked. It is not accessible by the application nor by an administrator.

It would be good to include it in the corresponding monitor event, but this would require a change to the socket monitor protocol, allowing an alternate or additional information to the address.

**Comment by sigiesec** [comment]
There are also other situations, where it might make sense to include an additional description, e.g. for the various ZMTP protocol errors.

**Comment by bluca** [comment]
@sigiesec I think you implemented this, right?

**Comment by sigiesec** [comment]
No... This is still unresolved. Do you think the monitoring event format can be modified/extended?

**Comment by sigiesec** [comment]
Maybe require opting in via a context option?

**Comment by bluca** [comment]
We have already so many options...
What if the new (or some of the current draft) events had additional multipart frames?

**Comment by sigiesec** [comment]
There are only few context options?

Currently, all events are specified as having the same format. I don't think the spec allows additional frames. So it is still incompatible if only done for new/draft events.

**Comment by bluca** [comment]
I know, I just think it gets confusing having options to change behaviour of an API - but it's fine if there's no other way

**Comment by bluca** [comment]
Did we do this as part of the event v2 API?

**Comment by sigiesec** [comment]
No, not yet, but the event v2 API enables implementing this

**Comment by bluca** [comment]
Great, reopening then
